src/lib/crypto/openssl/hash_provider/hash_evp.c

```diff
@@ -42,6 +42,11 @@ hash_evp(const EVP_MD *type, const krb5_crypto_iov *data, size_t num_data,
     size_t i;
     int ok;
 
+    /* Starting in OpenSSL 3.0, the evp_*() calls can fail if the appropriate
+     * provider is not loaded. */
+    if (type == NULL)
```

Comment:
I don't think this is quite right. Calls like `EVP_md4()` still will not fail. But if you attempt to use them then the initialisation will fail. So I suspect it must be the `EVP_DigestInit_ex()` call that was failing in the discussion in https://github.com/openssl/openssl/pull/10992